src/main/java/com/conveyal/datatools/manager/utils/MergeFeedUtils.java

```diff
@@ -0,0 +1,219 @@
+                "found %d of %d total stops that were incorrectly missing stop_code values. " +
+                "Note: \"special\" stops with location_type > 0 need not specify this value " +
+                "(%d special stops found in feed).",
+            stopsMissingStopCodeCount,
+            stopsCount,
+            specialStopsCount
+        );
+    }
+
+    /**
+     * Collect zipFiles for each feed version before merging tables.
+     * Note: feed versions are sorted by first calendar date so that future dataset is iterated over first. This is
+     * required for the MTC merge strategy which prefers entities from the future dataset over active feed entities.
+     */
+    public static List<FeedToMerge> collectAndSortFeeds(Set<FeedVersion> feedVersions, Auth0UserProfile owner) {
+        return feedVersions.stream()
+            .map(version -> {
+                try {
+                    return new FeedToMerge(version);
+                } catch (Exception e) {
+                    LOG.error("Could not create zip file for version: {}", version.version);
+                    ErrorUtils.reportToBugsnag(e, owner);
+                    return null;
+                }
+            })
+            // Filter out any feeds that do not have zip files (see above try/catch) and feeds that were never fully
+            // validated (which suggests that they would break things during validation).
+            .filter(Objects::nonNull)
+            .filter(
+                entry -> entry.version.validationResult != null
+                    && entry.version.validationResult.firstCalendarDate != null
+            )
+            // MTC-specific sort mentioned in above comment.
+            // TODO: If another merge strategy requires a different sort order, a merge type check should be added.
+            .sorted(
+                Comparator.comparing(
+                    entry -> entry.version.validationResult.firstCalendarDate,
+                    Comparator.reverseOrder())
+            ).collect(Collectors.toList());
+    }
+
+    /** Get all fields found in the feeds being merged for a specific table. */
+    public static Set<Field> getAllFields(List<FeedToMerge> feedsToMerge, Table table) throws IOException {
+        Set<Field> sharedFields = new HashSet<>();
+        // First, iterate over each feed to collect the shared fields that need to be output in the merged table.
+        for (FeedToMerge feed : feedsToMerge) {
+            CsvReader csvReader = table.getCsvReader(feed.zipFile, null);
+            // If csv reader is null, the table was not found in the zip file.
+            if (csvReader == null) {
+                continue;
+            }
+            // Get fields found from headers and add them to the shared fields set.
+            Field[] fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
+            sharedFields.addAll(Arrays.asList(fieldsFoundInZip));
+            csvReader.close();
```

Comment:
This should be wrapped in a try/finally as suggested previously by Evan.

Reply:
Thanks for the reminder on these ones. Handled in https://github.com/ibi-group/datatools-server/pull/376/commits/06b2156db917fa518ba09f157a55369277cc7aa7.